pkg/recipes/error_test.go

```diff
@@ -102,7 +102,7 @@ func TestGetRecipeErrorDetails(t *testing.T) {
 		},
 	}
 	for _, tc := range errorTests {
-		details := GetRecipeErrorDetails(tc.err)
+		details := GetErrorDetails(tc.err)
```

Comment:
Can you please add a test for when the error is `*azcore.ResponseError`?

Reply:
Added a test for ResponseError